langserver/hover.go

```diff
@@ -325,9 +433,14 @@ func (h *LangHandler) handleHoverGodef(ctx context.Context, conn jsonrpc2.JSONRP
 
 	loc := goRangeToLSPLocation(fset, res.Start, res.End)
 
+	// Handle builtin objects with invalid locations.
 	if loc.URI == "file://" {
-		// TODO: builtins do not have valid URIs or locations.
-		return &lsp.Hover{}, nil
+		_, node, _, _, _, _, _ := h.typecheck(ctx, conn, params.TextDocument.URI, params.Position)
```

Comment:
the only downside here is we try to avoid our typechecker in the `godef` based hover handler since it can be much slower and use more RAM. But given this will only be used on builtins its likely safe to fallback to this.